src/vivarium_census_prl_synth_pop/utilities.py

```diff
@@ -214,8 +214,7 @@ def vectorized_choice(
     cdf = np.cumsum(pmf)
 
     # for each p_i in probs, count how many elements of cdf for which p_i >= cdf_i
-    vect_find_index = np.vectorize(lambda p_i: (p_i >= cdf).sum())
-    chosen_indices = vect_find_index(probs)
+    chosen_indices = np.searchsorted(cdf[~pd.isnull(cdf)], probs, side="right")
```

Comment:
When would `cdf` be null?

Reply:
Sometimes the `weights` has NaNs in it (but non-NaN values for all the indices present in `options`). But, I am realizing this is only true of one call to `vectorized_choice` so it is probably easy to fix that one in this PR and remove this condition.

More broadly, the way this function interfaces between numpy and pandas seems suspect to me -- I think it depends on the index of `options` being sorted, when `options` is a Series -- but working that out and refactoring seemed potentially complex and out of scope for this performance improvement.

Note that none of this is new -- there were always NaNs in `cdf`, it was just hidden "better" by `(p_i >= cdf).sum()`.